multiqc/modules/mosaicatcher/mosaicatcher.py

```diff
@@ -0,0 +1,182 @@
+from multiqc.plots import bargraph, table
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        super(MultiqcModule, self).__init__(
+            name="MosaiCatcher",
+            anchor="mosaicatcher",
+            href="https://github.com/friendsofstrandseq/mosaicatcher",
+            info=(
+                "Mosaicatcher counts Strand-seq reads and classifies strand states "
+                "of each chromosome in each cell using a Hidden Markov Model."
+            ),
+            doi="10.1038/s41587-019-0366-x",
+        )
+        log_files = self.find_log_files("mosaicatcher", filehandles=False)
+        if not log_files:
+            log.warning("No log files found for MosaiCatcher module")
+        else:
+            print("Log files found:", log_files)
+
+        samples = dict()
+        for f in list(log_files):
+            if not f or "f" not in f:
+                log.warning("Malformed log file object: {}".format(f))
+            else:
+                print("Parsing log file:", f["fn"])
+                samples = self._parse_samples(f, samples)
+                self.add_data_source(f)
+
+        self._samples = samples
+
+        self._add_table(samples)
+        self._add_coverage_plot(samples)
+
+        self.write_data_file(self._samples, "multiqc_mosaicatcher")
+        log.info("Found {} samples".format(len(samples)))
+
+    def _add_table(self, samples):
+        self.add_section(
+            plot=table.plot(data=samples, headers=self._setup_headers()),
+        )
+
+    def _add_coverage_plot(self, samples):
+        coverage_plot = bargraph.plot(
+            data=samples,
+            cats=["mapq", "dupl", "good", "reads2", "suppl"],
+            pconfig={
+                "id": "mosaicatcher-coverage",
+                "title": "MosaiCatcher: coverage",
+                "height": 1024,
+                "ymax": 4e6,
+                "ylab": "Cells",
+            },
+        )
+        self.add_section(
+            name="Average coverage per sample",
+            anchor="mosaicatcher-coverage-id",
+            description="Average coverage per sample",
+            plot=coverage_plot,
+        )
+
+    def _setup_headers(self):
+        headers = OrderedDict()
+        for k in list(self._samples.values())[0]:
+            headers[k] = {"title": k, "hidden": True}
+
+        headers["sample"] = {
+            "title": "sample",
+            "description": "Sample (has multiple cells)",
+            "hidden": True,
+        }
+        headers["cell"] = {
+            "title": "cell",
+            "description": "Name of the cell.",
+            "hidden": False,
+        }
+        headers["mapped"] = {
+            "title": "mapped",
+            "max": 4e6,
+            "description": "Total number of reads seen",
+            "scale": "RdYlGn",
+            "hidden": False,
+        }
+        headers["suppl"] = {
+            "title": "suppl",
```

Comment:
Please capitalize the column names (`Cell`, `Mapped`, etc), replace the underscores with spaces.

Should also add "namespace": "MosaiCatcher" field into the headers.

Reply:
Do you mean the title key into the nested dictionnary or the headers dict keys?